docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -629,15 +613,31 @@ def table1() -> Output[DataFrame]:
 This works even if you're not returning an object from your decorated function:
 
 ```python file=/concepts/assets/asset_materialization_metadata_none.py
-from dagster import Output, asset
+from dagster import AssetExecutionContext, asset
 
 
 @asset
-def table1() -> Output[None]:
+def table1(context: AssetExecutionContext) -> None:
     ...  # write out some data to table1
-    return Output(None, metadata={"num_rows": 25})
+    context.add_output_metadata(metadata={"num_rows": 25})
```

Comment:
not 100% sure this is what we want to recommend, but ti's the approach we are taking in the tutorial for when an asset had no return value, but you want to attach metadata. We should update this to whatever the outcome of this [discussion](https://github.com/dagster-io/dagster/discussions/15392) is